docs/reference/search.md

```diff
@@ -0,0 +1,102 @@
+
+Check the [official documentation](https://opensearch.org/docs/2.0/opensearch/ux/) and [Reference](#reference) for more context on the `edge_ngram` filter and custom analyzers.
+
+### When to Use the Suggest API
+
+- **Typo Tolerance & Auto-completion:** Helps correct typos (using `fuzziness` at search time analyzing) and completes partial inputs.
+- **Large, Diverse Datasets:** Useful for datasets with a wide variety of terms, like names or titles.
+- **Pre-query Optimization:** Reduces unnecessary searches by suggesting relevant terms.
+
+### When Not to Use the Suggest API
+
+- **Small or Specific Datasets:** Less beneficial for well-defined datasets.
+- **Performance Constraints:** Because the suggest API creates large amounts of token using the `edge_ngram` filter, it is important to observe how it affects the index size.
+  - A reasonable trade-off might involve an index size increase of up to 20-30% if it significantly improves search speed and relevance.
+  - A 10-20% improvement in response times might justify a moderate increase in index size.
+
+For more information check the [official documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-disk-usage.html).
+
+## Key Considerations for Customizing Index Mappings
+
+### Size
+
+- Field Type Selection: Use lightweight field types (e.g., keyword over text where appropriate) to minimize index size.
+- Usage of custom analyzers or filters should be as limited as possible to prevent index bloating.
+
+### Speed
+
+- Search Performance: Kepeping size in mind, apply custom analyzers that include `edge_ngram` filter for providing quick suggestions, and optimize for frequently queried fields to enhance search speed.
+- Analyzer and filter selection: Configure only when necessary to improve search time.
+
+## Fine tuning the search
+
+- **Relevance Adjustment:** Boosting affects the relevance score of documents. A higher boost value means a stronger influence on the search ranking. Determine which fields are most critical for your search relevance (e.g., titles, authors, keywords).
+
+Boosting of field(s) can be done by using the caret operator **(^)** followed by a number.
+Examples: `name^100` will boost the name field by a factor of 100.
+Asterisk **(\*)** can be used to apply boosting to all the subfields. `i18n_titles.*^50`
+
+- **Balance and Tuning:** Use boosting judiciously to avoid skewing results too heavily towards particular fields. Assign boost factors based on the importance of each field. Higher values increase the influence of matches in that field.
```

Comment:
```suggestion
Boosting affects the relevance score of documents. A higher boost value means a stronger influence on the search ranking. Determine which fields are most critical for your search relevance (e.g., titles, authors, keywords).

- **Relevance Adjustment:** Boosting of field(s) can be done by using the caret operator **(^)** followed by a number. For example:
    * `name^100` will boost the name field by a factor of 100.
    * Asterisk **(\*)** can be used to apply boosting to all the subfields. `i18n_titles.*^50`

- **Balance and Tuning:** Use boosting judiciously to avoid skewing results too heavily towards particular fields. Assign boost factors based on the importance of each field. Higher values increase the influence of matches in that field.
```

I think this works better having an introductory sentence first